metadata-ingestion/src/datahub/ingestion/source/vertica/common.py

```diff
@@ -0,0 +1,1628 @@
+                    TypeClass = NullTypeClass
+                    break
+
+        if TypeClass is None:
+            sql_report.report_warning(
+                dataset_name, f"unable to map type {column_type!r} to metadata schema"
+            )
+            TypeClass = NullTypeClass
+
+        return SchemaFieldDataType(type=TypeClass())
+
+    def get_schema_metadata(
+        sql_report: SQLSourceReport,
+        dataset_name: str,
+        platform: str,
+        foreign_keys: Optional[List[ForeignKeyConstraint]] = None,
+        canonical_schema: List[SchemaField] = [],
+    ) -> SchemaMetadata:
+
+        schema_metadata = SchemaMetadata(
+            schemaName=dataset_name,
+            platform=make_data_platform_urn(platform),
+            version=0,
+            hash="",
+            platformSchema=MySqlDDL(tableSchema=""),
+            fields=canonical_schema,
+        )
+        if foreign_keys is not None and foreign_keys != []:
+            schema_metadata.foreignKeys = foreign_keys
+
+        return schema_metadata
+
+    def gen_schema_key(self, db_name: str, schema: str) -> PlatformKey:
+        try:
+            all_properties_keys = dict()
+            for inspector in self.get_inspectors():
+
+                all_properties_keys = inspector._get_properties_keys(db_name , schema, level='schema')
+
+            return SchemaKeyHelper(
+                database=db_name,
+                schema=schema,
+                platform=self.platform,
+                instance=self.config.platform_instance,
+                backcompat_instance_for_guid=self.config.env,
+
+                numberOfProjection=all_properties_keys.get("projection_count", ""),
+                udxsFunctions=all_properties_keys.get("udx_list", ""),
+                UDXsLanguage=all_properties_keys.get("Udx_langauge", ""),
+            )
+        except Exception as e:
+            print("Hey something went wrong, while gettting schema in gen schema key")
+
+    def gen_database_key(self, database: str) -> PlatformKey:
+        try:
+            all_properties_keys = dict()
+            for inspector in self.get_inspectors():
+                all_properties_keys = inspector._get_properties_keys(database , "schema", level='database')
+
+            return DatabaseKeyHelper(
+                database=database,
+                platform=self.platform,
+                instance=self.config.platform_instance,
+                backcompat_instance_for_guid=self.config.env,
+
+
+                clusterType=all_properties_keys.get("cluster_type"),
+                clusterSize=all_properties_keys.get("cluster_size"),
+                subClusters=all_properties_keys.get("Subcluster"),
+                communalStoragePath=all_properties_keys.get("communinal_storage_path"),
+            )
+        except Exception as e:
+            traceback.print_exc()
+            print("Hey something went wrong, while gettting Generation of database key")
```

Comment:
use report.report_failure

Reply:
thanks for pointing it out, we have fixed it and the updated code will be in the next commit.